python/pyarrow/tests/test_fs.py

```diff
@@ -439,6 +438,28 @@ def allow_append_to_file(request, filesystem_config):
     return filesystem_config['allow_append_to_file']
 
 
+@pytest.fixture(
+    params=[
+        pytest.lazy_fixture('builtin_pickle'),
+        pytest.lazy_fixture('cloudpickle')
+    ]
+)
+def pickle(request):
+    return request.param
```

Comment:
We have other pickling tests which could benefit from this. Perhaps share this fixture with all test modules and make sure it gets used everywhere possible?
(also perhaps rename it to `picklemodule` for less confusion, though YMMV)

Reply:
Will do! I was thinking the same, but thought it might be better in a different PR. Now that you explicitly mention it, it is probably best to test cloudpickle everywhere as part of this cython upgrade.

Reply:
The commit to implement this is pretty big (I already pushed it to this PR). I went ahead and implemented in a separate PR and can rebase on top of it after it merges. This has the additional benefit of testing the CI process with Cython 0.29.X, since I haven't added the Cython < 3 CI job yet.

https://github.com/apache/arrow/pull/37255